### PR TITLE
Feat/grappling hook

### PR DESCRIPTION
**Pull request by HoJunHao2000**
Incorporated grappling hook system with new ECS structure.

There may still be bugs in the game. Do test it out and tell me

https://github.com/cs3217-2324/StarDash/assets/84698566/0cbf2712-9e53-4234-a2d9-444ca9b14091

### CODE REVIEW

File: star-dash/star-dash/GameEngine/Components/GrappleHookOwnerComponent.swift [hunk 1]
```diff
@@ -0,0 +1,21 @@
+//
+//  GrappleHookOwnerComponent.swift
+//  star-dash
+//
+//  Created by Ho Jun Hao on 28/3/24.
+//
+
+import Foundation
+
+class GrappleHookOwnerComponent: Component {
```
Comment (ruihan00) [inline star-dash/star-dash/GameEngine/Components/GrappleHookOwnerComponent.swift:10]: Why do we need OwnerComponent if grapple hook already has the owner id?

**Comment by ruihan00** [comment]
nice work! but expected behaviour for grapple is only available if player is already jumping

**Review by ruihan00 — APPROVED** [review]
Minor changes but very nice work

File: star-dash/star-dash/GameEngine/Components/GrappleHookOwnerComponent.swift (hunk 1 above)
Comment (ruihan00) [reply star-dash/star-dash/GameEngine/Components/GrappleHookOwnerComponent.swift:10]: sorry, reviewed again and realised there's a grapple hook entity, but would a ownerPlayerId field in the component make more sense?

Comment (HoJunHao2000) [reply star-dash/star-dash/GameEngine/Components/GrappleHookOwnerComponent.swift:10]: sure. changed

**Pull request merged**